src/Git/GitRepository.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHP\Git;
+
+use Gitonomy\Git\Diff\Diff;
+use Gitonomy\Git\Repository;
+use Gitonomy\Git\WorkingCopy;
+use GrumPHP\Locator\GitRepositoryLocator;
+
+/**
+ *
+ * Small wrapper on top of the gitonomy repository class.
+ * This makes it possible to lazy load the repository and add additional features on top of that library.
+ */
+class GitRepository
+{
+    /**
+     * @var GitRepositoryLocator
+     */
+    private $repositoryLocator;
+
+    /**
+     * @var array
+     */
+    private $repositoryOptions;
+
+    /**
+     * @var ?Repository
```

Comment:
@veewee, what's the best way to typehit: `null|Repository` or `?Repository`?

Reply:
`null|Repository` works best in most IDEs I guess ...